pkg/config/config.go

```diff
@@ -1445,9 +1450,80 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 	// setTracemallocEnabled *must* be called before setNumWorkers
 	warnings.TraceMallocEnabledWithPy2 = setTracemallocEnabled(config)
 	setNumWorkers(config)
+	setupFipsEndpoints(config)
 	return &warnings, nil
 }
 
+// setupFipsEndpoints overwrite the Agent endpoint for outgoing data to be sent to the local FIPS proxy. The local FIPS
+// proxy will be in charge of forwarding data to the Datadog backend following FIPS standard. Starting from
+// fips.port_range_start we will assign a dedicated port per product (metrics, logs, traces, ...).
+func setupFipsEndpoints(config Config) {
+	// Assuming that fips.port_range_start is the default 3833, each port is dedicated to a specific data type:
+	//
+	// port_range_start: HAProxy stats
+	// port_range_start + 1:  metrics
+	// port_range_start + 2:  traces
+	// port_range_start + 3:  profiles
+	// port_range_start + 4:  processes
+	// port_range_start + 5:  logs
+	// port_range_start + 6:  databases monitoring metrics
+	// port_range_start + 7:  databases monitoring samples
+	// port_range_start + 8:  network devices metadata
+	// port_range_start + 9:  network devices snmp traps (unused)
+	// port_range_start + 10: instrumentation telemetry
+	// port_range_start + 11: appsec events (unused)
+
+	if !config.GetBool("fips.enabled") {
+		log.Debug("fips mode is disabled")
+		return
+	}
+
+	const (
```

Comment:
If I added a new item in the middle of this list, or "tidied up" by sorting it, how bad would that be?  Should these have explicit numbers, or a comment indicating items should only be added at the end?

Reply:
I'd +1 to explicit numbers. If we remove one of the entries in the future, we shouldn't have to renumber the items. I'm also a bit wondering if these constants should be in a separate go file or at the top since they're effectively static constants.